### PR TITLE
Made the `Pagination`  component headless

### LINKED ISSUE #1888: Make the `@yamada-ui/pagination` component headless
script: make-component-headless
package: @yamada-ui/pagination

### Description
Currently, the `@yamada-ui/pagination` component has minimal styles set within the component's source code.
We want to make this style transferable to the theme and make the component headless, enabling more flexible UI customization.

### Problem Statement/Justification
By making the `@yamada-ui/pagination` component headless, we can achieve more flexible styling tailored to the project.

### Proposed Solution or API
The styles within the component are mostly defined by `css: CSSUIObject` and set to a prop named `__css`. We will transfer these styles to the `packages/theme/src/components`.

The code below is just an example and not the actual code.

`packages/components/input/src/input-group.tsx`
```diff
const css: CSSUIObject = {
- width: "100%",
- display: "flex",
- position: "relative",
  ...styles.container,
}
```

`packages/theme/src/components/input.ts`
```diff
baseStyle: {
  container: {
+   width: "100%",
+   display: "flex",
+   position: "relative",
  },
  field: {
    width: "100%",
    minWidth: 0,
    outline: 0,
    position: "relative",
    appearance: "none",
    transitionProperty: "common",
    transitionDuration: "normal",
    _disabled: {
      opacity: 0.4,
      cursor: "not-allowed",
    },
  },
  element: {
    color: ["blackAlpha.600", "whiteAlpha.700"],
  },
},
```

> [!TIP]
>
> In cases where the styles within the component are dynamically changed, there may be instances where it is not possible to transfer the styles to the `theme`.
> In such cases, it needs to be discussed in this issue. Please feel free to send comments to this issue.

### Additional Information
The `@yamada-ui/pagination` component exists in `packages/components/pagination/src`. You need to check all the components in this folder and if they contain styles internally, you need to modify them.

This change could have a significant impact on projects using Yamada UI, so we will work on it within the scope of `v2.0`.
Please create the branch for working on this issue from the remote branch `v2.0`, and direct your pull requests to the `v2.0` branch as well.

Please select `major` for the changeset and create a pull request.

### PR DESCRIPTION
**Pull request by Haru-0001**
Closes #1888

## Description

Made the Pagination component headless

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Review by koralle — APPROVED** [review]

File: packages/components/pagination/src/pagination.tsx
Comment (fffk10) [inline packages/components/pagination/src/pagination.tsx:1]: I feel like this is also necessary, but is it unnecessary?

https://github.com/yamada-ui/yamada-ui/blob/8047466fc462c845dca7df2fefb5dd04414e584a/packages/components/pagination/src/pagination.tsx#L193-L199

**Review by fffk10 — CHANGES_REQUESTED** [review]

File: packages/components/pagination/src/pagination.tsx
Comment (Haru-0001) [reply packages/components/pagination/src/pagination.tsx:1]: I had lost track of it. I will correct it.

Comment (Haru-0001) [reply packages/components/pagination/src/pagination.tsx:1]: @fffk10 @koralle 
I have made the change. Could you please review it?

**Review by nakayan5 — APPROVED** [review]

File: packages/components/pagination/src/pagination.tsx
Comment (fffk10) [reply packages/components/pagination/src/pagination.tsx:1]: LGTM!

**Review by fffk10 — APPROVED** [review]

**Pull request merged**